### PR TITLE
add chat with particular note, and prompt template

### PR DESCRIPTION
**Pull request by chuyouchia**
Adding the component for now, to be used later with the ask-file format

https://github.com/reorproject/reor/assets/58682208/a958d470-458f-4377-a4e0-25f3e3912de0

### CODE REVIEW

File: src/components/Chat/Chat.tsx [hunk 1]
```diff
@@ -184,7 +184,7 @@ const ChatWithLLM: React.FC = () => {
           <FiRefreshCw className="text-white" /> {/* Icon */}
         </div>
       </div>
-      <div className="flex-1 overflow-auto p-4 pt-0 bg-transparent">
+      <div className="flex flex-col overflow-auto p-4 pt-0 bg-transparent h-full">
```
Comment (chuyouchia) [inline src/components/Chat/Chat.tsx:187]: These are just layout changes for the prompt templates to be situated near to the input text area when its added later

Comment (chuyouchia) [reply src/components/Chat/Chat.tsx:187]: Should note that in absence of the additional prompt template, nothing changes about the layout behavior
image

File: src/components/Chat/Chat.tsx [hunk 2]
```diff
@@ -250,13 +274,45 @@ const ChatWithLLM: React.FC = () => {
...
+              </button>
+              </MenuHandler>
+              <MenuList placeholder="" className="bg-slate-600">
+                {ASK_OPTIONS.map((option, index) => {
+                  return (
+                    <MenuItem key={index} placeholder="" 
+                      className="bg-slate-600 border-none h-full w-full hover:bg-slate-700 cursor-pointer text-white text-left" 
+                      onClick={() => setAskText(option)}>
+                        {option}
+                    </MenuItem>
```
Comment (chuyouchia) [inline src/components/Chat/Chat.tsx:309]: Now we have notes mode:
1. Ask specific note
2. Template prompts for those notes
3. Soon to directly connect with current file interfaced.

https://github.com/reorproject/reor/assets/58682208/1f6f2cc1-c853-4de6-a3f6-9fe7b8d5f6ae

Next up is allowing users to customize their own prompts

Comment (samlhuillier) [reply src/components/Chat/Chat.tsx:309]: nice ill add in the logic to filter for single file

**Review by samlhuillier — APPROVED** [review]

**Pull request merged**